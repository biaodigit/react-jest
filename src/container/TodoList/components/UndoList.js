import React, { Component } from 'react';
import '../style.css'

class UndoList extends Component {

    render() {
        const { list, deleteItem, changeStatus, handleBlur, valueChange } = this.props
        return (
            <div className="undo-list">
                <div className="undo-list-title">
                    正在进行
                    <div data-test="count" className="undo-list-count">{list.length}</div>
                </div>
                <ul className="undo-list-content">
                    {list.map((item, index) => (
                        <li key={index} data-test="list-item" className="undo-list-item" onClick={() => changeStatus(index)}>
                            {item.focus ? <input className="undo-list-input" data-test="input" onBlur={() => handleBlur(index)
                            } value={item.value} onChange={(e) => valueChange(index, e.target.value)} /> : item.value}
                            <span onClick={(e) => {
                                e && e.stopPropagation()
                                deleteItem(index)
                            }} data-test="del-item" className="undo-list-delete">-</span>
                        </li>
                    ))}
                </ul>
            </div>
        )
    }
}

export default UndoList;
