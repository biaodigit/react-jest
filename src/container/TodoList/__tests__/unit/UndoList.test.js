import React from 'react';
import { shallow } from 'enzyme';
import { findTestWrapper } from '../../../../utils/testUtils'
import UndoList from '../../components/UndoList'

let wrapper;
describe('test undolist component', () => {
    beforeEach(() => {
    })

    it('render empty undolist', () => {
        // expect(wrapper).toMatchSnapshot()
        wrapper = shallow(<UndoList list={[]} />)
        const countEl = findTestWrapper(wrapper, 'count')
        const listItems = findTestWrapper(wrapper, 'list-item')
        expect(countEl.text()).toEqual('0')
        expect(listItems.length).toEqual(0)
    })

    it('render unempty undolist', () => {
        // expect(wrapper).toMatchSnapshot()
        wrapper = shallow(<UndoList list={['test1', 'test2']} />)
        const countEl = findTestWrapper(wrapper, 'count')
        const listItems = findTestWrapper(wrapper, 'list-item')
        expect(countEl.text()).toEqual('2')
        expect(listItems.length).toEqual(2)
    })

    it('if undolist unempty,can delete', () => {
        const fn = jest.fn()
        wrapper = shallow(<UndoList deleteItem={fn} list={['test1', 'test2']} />)
        const delItems = findTestWrapper(wrapper, 'del-item')
        delItems.at(1).simulate('click')
        expect(fn).toHaveBeenLastCalledWith(1)
    })

})
