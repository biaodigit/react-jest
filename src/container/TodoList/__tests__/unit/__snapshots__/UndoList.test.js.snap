// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test undolist component render snapshot 1`] = `
<div
  className="undo-list"
>
  <div
    className="undo-list-title"
  >
    正在进行
    <div
      className="undo-list-count"
      data-test="count"
    >
      2
    </div>
  </div>
  <ul
    className="undo-list-content"
  >
    <li
      className="undo-list-item"
      data-test="list-item"
      key="0"
      onClick={[Function]}
    >
      test1
      <span
        className="undo-list-delete"
        data-test="del-item"
        onClick={[Function]}
      >
        -
      </span>
    </li>
    <li
      className="undo-list-item"
      data-test="list-item"
      key="1"
      onClick={[Function]}
    >
      <input
        className="undo-list-input"
        data-test="input"
        onBlur={[Function]}
        onChange={[Function]}
        value="test2"
      />
      <span
        className="undo-list-delete"
        data-test="del-item"
        onClick={[Function]}
      >
        -
      </span>
    </li>
  </ul>
</div>
`;
